### PR TITLE
feat: create action for triaging labeled issues

### PR DESCRIPTION
**Pull request by matthewgallo**
Contributes to #4803 

Adds the prototype action I created [here](https://github.com/matthewgallo/ibm-products-vite-template/blob/main/.github/workflows/add-label-move-to-area.yml) into our repo. This action triages labeled issues by adding them to our project board and applying the specified field/field option values. In this case, issues labeled `area: typescript` will be moved into "Area: 🟦 TypeScript" in our project.

Discussed with @tay1orjones about potentially including the reusable workflow in a [shared repo](https://github.com/carbon-design-system/.github) for other teams to use as well, but want to make sure it works here first.

If we want to add this same thing for other labels we should be able to easily by updating `.github/workflows/triage-labeled-issues.yml`. The [Graphql API Explorer](https://docs.github.com/en/graphql/overview/explorer) really helped with being able to see the response and to make sure that the `query` was properly structured.

#### What did you change?
Created two new workflow files
```
.github/workflows/move-issue-to-project-and-set-fields.yml // more generic reusable workflow file
.github/workflows/triage-labeled-issues.yml
```
#### How did you test and verify your work?
[In test repo](https://github.com/matthewgallo/ibm-products-vite-template/tree/main/.github/workflows)

### CODE REVIEW

File: .github/workflows/triage-labeled-issues.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  move-ts-issues:
+    if: ${{contains(github.event.label.name, format('area{0} typescript', ':'))}}
+    uses: ./.github/workflows/move-issue-to-project-and-set-fields.yml
+    with:
+      field: 'Area'
+      field_option: '🟦 Typescript'
+      project_number: '65'
+    secrets: inherit
+  move-migration-issues:
+    if: ${{contains(github.event.label.name, format('area{0} migration ➡️', ':'))}}
```
Comment (matthewgallo) [inline .github/workflows/triage-labeled-issues.yml:17]: Not sure how that arrow is going to work but that's how the emoji pasted when I got it from the label 🤔

Comment (elycheea) [reply .github/workflows/triage-labeled-issues.yml:17]: Fwiw, it did show up correctly in my email. So _probably_ fine? 😁

Comment (matthewgallo) [reply .github/workflows/triage-labeled-issues.yml:17]: Ah interesting, I've found that some emojis render differently in vscode vs github so maybe there is some kind of mapping that happens. Learning a lot about emojis with this PR 😂

File: .github/workflows/triage-labeled-issues.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    uses: ./.github/workflows/move-issue-to-project-and-set-fields.yml
+    with:
+      field: 'Area'
+      field_option: '🗺 Migration'
+      project_number: '65'
+    secrets: inherit
+  move-a11y-issues:
+    if: ${{contains(github.event.label.name, format('type{0} ally ♿', ':'))}}
```
Comment (elycheea) [inline .github/workflows/triage-labeled-issues.yml:25]: Should this one actually be a11y and not ally?

```suggestion
    if: ${{contains(github.event.label.name, format('type{0} a11y ♿', ':'))}}
```

File: .github/workflows/triage-labeled-issues.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      field_option: '🟦 Typescript'
+      project_number: '65'
+    secrets: inherit
+  move-migration-issues:
+    if: ${{contains(github.event.label.name, format('area{0} migration ➡️', ':'))}}
+    uses: ./.github/workflows/move-issue-to-project-and-set-fields.yml
+    with:
+      field: 'Area'
+      field_option: '🗺 Migration'
+      project_number: '65'
```
Comment (elycheea) [inline .github/workflows/triage-labeled-issues.yml:22]: Not sure if it’s worth storing this as an environment variable[^1]. I think adding more labels to our workflows will be at least somewhat manual so maybe lower value.

[^1]: https://docs.github.com/en/actions/learn-github-actions/variables#defining-environment-variables-for-a-single-workflow

File: .github/workflows/triage-labeled-issues.yml (hunk 2 above)
Comment (matthewgallo) [reply .github/workflows/triage-labeled-issues.yml:25]: Yes 👍

**Review by elycheea — APPROVED** [review]

File: .github/workflows/triage-labeled-issues.yml (hunk 3 above)
Comment (matthewgallo) [reply .github/workflows/triage-labeled-issues.yml:22]: I did try, but got an error about using environment variables for [workflow_call inputs](https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_callinputs).
image

**Pull request merged**